src/ripple/app/paths/impl/BookStep.cpp

```diff
@@ -873,9 +898,19 @@ BookStep<TIn, TOut, TDerived>::fwdImp (
 
         if (offersConsumed >= maxOffersToConsume_)
         {
-            // Too many iterations, mark this strand as dry
-            cache_.emplace (beast::zero, beast::zero);
-            return {beast::zero, beast::zero};
+            // Too many iterations, mark this strand as inactive (dry)
+            if (!afView.rules().enabled(fix1515))
+            {
+                // Don't use the liquidity
+                cache_.emplace(beast::zero, beast::zero);
+                return {beast::zero, beast::zero};
+            }
+            else
```

Comment:
You `return` from the above block. The `else` is unnecessary. Same for line 744.

Reply:
Fixed.